### PR TITLE
Fix timezone placement in add_missing_timezones()

### LINKED ISSUE #844: [BUG] Missing timezones are added to the end of the file but should be at the start
## Describe the bug

Adding missing timezones puts them last. They should be first!

## To Reproduce

```
from icalendar import Calendar, Event
from datetime import datetime
from pathlib import Path
import uuid
from zoneinfo import ZoneInfo

calendar = Calendar()
calendar.add("VERSION", "2.0")
calendar.add("PRODID", "my calendar program")

event = Event()
event.add("UID", uuid.uuid4())
event.start = datetime(2026, 3, 19, 12, 30, tzinfo=ZoneInfo("Europe/London"))
event.DTSTAMP = datetime.now(ZoneInfo("UTC"))

event.add("SUMMARY", "This is my first event!")

calendar.add_component(event)

calendar.add_missing_timezones()
Path("export.ics").write_bytes(calendar.to_ical())

...
```
## Expected behavior

The timezone should be first

## Environment

 - [ ] OS:       Ubuntu                        
 - [ ] Python version: 3.12           
 - [ ] `icalendar` version:   6.3.0   

## Additional context

- [ ] I tested it with the latest version `pip3 install https://github.com/collective/icalendar.git`
- [ ] I attached the ICS source file or there is no ICS source file

### PR DESCRIPTION
**Pull request by SashankBhamidi**
Fixes #844 by modifying Calendar.add_missing_timezones() to insert VTIMEZONE components at the beginning of the calendar instead of appending them to the end.

## Problem
The add_missing_timezones() method was placing VTIMEZONE components after VEVENT and other components that reference them. This violates standard iCalendar practices where VTIMEZONE components should appear before components that reference them.

## Solution
- Modified the insertion logic to place new VTIMEZONE components after any existing VTIMEZONE components but before other component types
- Preserves existing timezone components in their current positions
- Maintains backward compatibility with all existing functionality

## Changes
- Modified: src/icalendar/cal/calendar.py - Updated add_missing_timezones() method
- Added: src/icalendar/tests/test_issue_844_timezone_placement.py - Comprehensive test coverage
- Updated: CHANGES.rst and docs/credits.rst per contributing guidelines

## Test Coverage
- Single and multiple timezone scenarios
- Existing timezone preservation  
- Edge cases (empty calendar, no missing timezones)
- Serialized output validation
- Forward reference compatibility
- All existing tests continue to pass (755 tests total)

----
📚 Documentation preview 📚: https://icalendar--878.org.readthedocs.build/

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -41,6 +41,7 @@ New features:
 Bug fixes:
 
 - Fix invalid calendar: Parsing a date with TZID results in a datetime to not loose the timezone. See `Issue 187 <https://github.com/collective/icalendar/issues/187>`_.
+- Fix timezone placement in add_missing_timezones(): VTIMEZONE components now appear before VEVENT and other components that reference them. See `Issue 844 <https://github.com/collective/icalendar/issues/844>`_.
```
Comment (stevepiercy) [inline CHANGES.rst]:
```suggestion
- Fix timezone placement in ``add_missing_timezones()``. ``VTIMEZONE`` components now appear before ``VEVENT`` and other components that reference them. See `Issue 844 <https://github.com/collective/icalendar/issues/844>`_.
```

**Review by stevepiercy — APPROVED** [review]
This still needs a technical review. docstrings and news look good.

File: src/icalendar/tests/test_issue_844_timezone_placement.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+    # This should still parse correctly due to existing forward reference handling
+    calendar = Calendar.from_ical(ical_content)
+    assert len(calendar.events) == 1
+    assert len(calendar.timezones) == 1
+    assert calendar.timezones[0].tz_name == "Europe/London"
+
+
+def test_timezone_placement_without_pytz():
+    """Test timezone placement. Works even if pytz is not available."""
```
Comment (niccokunzmann) [inline src/icalendar/tests/test_issue_844_timezone_placement.py:187]: Hi, this is also done in the CI with 

```
tox -e nopytz
```

File: src/icalendar/tests/test_issue_844_timezone_placement.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+    calendar.add_missing_timezones()
+
+    components = [comp.name for comp in calendar.subcomponents]
+    assert components[:2] == ["VTIMEZONE", "VTIMEZONE"]
+    assert components[2:] == ["VEVENT", "VEVENT"]
+
+    # Verify both timezones are present (existing + new)
+    timezone_names = {tz.tz_name for tz in calendar.timezones}
+    assert timezone_names == {"Europe/Berlin", "America/New_York"}
```
Comment (niccokunzmann) [inline src/icalendar/tests/test_issue_844_timezone_placement.py:91]: Thanks, this PR looks really good.

**Review by niccokunzmann — APPROVED** [review]
Thank you very much for this thorough work!
